File: bcw5/php/Vector/Vector.php
<?php    
    class Vector {
        private $x;
        private $y;

        public function Vector($x=0, $y=0) {
            $this->x = $x;
            $this->y = $y;
        }

        public function getX() {
            return $this->x;
        }

        public function getY() {
            return $this->y;
        }

        public function setX($x) {
            $this->x = $x;
        }

        public function setY($y) {
            $this->y = $y;
        }

        public function len($other) {
            return hypot($this->x-$other->x, $this->y-$other->y);
        }

        public function __toString() {
            return sprintf("(%s, %s)", $this->x, $this->y);
        }

        public function add($other) {
            $this->x += $other->x;
            $this->y += $other->y;
        }

        public function sub($other) {
            $this->x -= $other->x;
            $this->y -= $other->y;
        }
    }

    function isEqual($x, $y) {
        if ( $x == $y ) {
            echo $x . " is equal to " . $y . PHP_EOL;
        }
    }

    function isNotEqual($x, $y) {
        if ( $x != $y ) {
            echo $x . " isn`t equal " . $y . PHP_EOL;
        }
    }

    $a = new Vector(0, 5);
    $b = new Vector(3, 19);
    $c = new Vector(3, 78);
    $d = $a;
    $e = clone $a;

    echo "a: " . $a . PHP_EOL;
    echo "b: " . $b . PHP_EOL;
    echo "c: " . $c . PHP_EOL;

    isEqual($a, $b);
    isEqual($b, $c);

    isNotEqual($a, $b);
    isNotEqual($a, $d);

    $a->add($c);

    echo "a: " . $a . PHP_EOL;

    echo $a->len($b) . PHP_EOL;
    echo $b->len($c) . PHP_EOL;
?>

File: bcw4/java/Basic/Vector/Vector.java
public class Vector {
    private double x;
    private double y;

    public Vector() {
        this.x = 0;
        this.y = 0;
    }

    public Vector(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public double getY() {
        return y;
    }

    public void setX(double value) {
        x = value;
    }

    public void setY(double value) {
        y = value;
    }

    public double len() {
        return Math.hypot(x, y);
    }

    public boolean equals(Vector other) {
        return x == other.x && y == other.y;
    }

    public boolean notEquals(Vector other) {
        return !equals(other);
    }

    public void addTo(Vector other){
        x += other.x;
        y += other.y;
    } 

    public void subtractFrom(Vector other) {
        x -= other.x;
        y -= other.y;
    }

    public Vector sum(Vector other) {
        Vector temp = new Vector();

        temp.x = x + other.x;
        temp.y = y + other.y;
        return temp;
    } 

    public Vector sub(Vector other) {
        Vector temp = new Vector();

        temp.x = x - other.x;
        temp.y = y - other.y;
        return temp;
    }

    public String toString() {
        return "(" + x + ", " + y + ")";
    }

    public void print() {
        System.out.format("(%.2f, %.2f)%n", x, y);
    }
}